salt/fileclient.py

```diff
@@ -1278,8 +1278,13 @@ def hash_and_stat_file(self, path, saltenv='base'):
                 'saltenv': saltenv,
                 'cmd': '_file_find'}
         fnd = self.channel.send(load)
-        stat_result = fnd.get('stat')
-        return hash_result, stat_result
+
+        try:
+            fnd_mode = fnd.get('stat', [])[0]
```

Comment:
Nope. This won't work. The fileclient expects the list-ified stat_result and checks the first element in the list.